### ISSUE TITLE
Warrior charge still falling through ground on live

### ISSUE THREAD

**Issue by Break977** [dalaranwow/dalaran-wow#4274]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Game server.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behavior**: when charging a target, especially on hills, I fall below the map (or if there is a cave system below me) i will fall into that

**Expected behavior**: Just read the tooltip for the spell Charge, thats what should happen.

**Steps to reproduce the problem**:

1. Create warrior
2. Target mob (try Westfall or Elwynn Forest) charge while on anything other than flat ground
3. Watch yourself falling under map

**Include proofs for this behavior**

"Issues without proofs will be closed and tag On Hold will be added until the proof is provided." -- I dont record while playing, this bug doesnt need any proof outside of a dev trying it for themselves, besides, just look at all the other reports of the same thing.

**Include Screenshots from the issue if necessary**

**Comment by Rushor** [comment]
https://github.com/TrinityCore/TrinityCore/issues/11526

**Comment by Rushor** [comment]
already fixed on PTR:

Core/Mmaps: Charge ignoring z position of target (#17516)

* commit: fda9f1a1367dfe5f7829b8223afa2057429a8020

The issue was fixed by a developer and will now be tested on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Only once the issue was marked with "Fixed - On Live Server", you can find the fix on you server.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by Aridam** [comment]
Not sure which report to bump as this issue has multiple of them, so I'm gonna go with this one: Confirming this as fixed but only to an extent -- bumpy terrain in general which lets you Charge negating most-to-all pathing gets you through textures, and most often end in you falling through them, which often results in death (and then you need to "/reload" or use anything similar to be popped back up in the world at the nearest Graveyard) -- note however that _some_ still abides by pathing rules and you either follow it as-if pinpoint-accurately, or you get the error "Out of range" which is also correct, so the fix has indeed greatly-reduced the occurrence of this, but there's still more of it happening that's left.

Furthermore, this issue still suffers greatly from water-based Charging -- both deep-water Charging _and_ on the surface where the ground ends; most often you won't be able to use any Charge mechanics because of the "No path found" error which is invalid as there are clear paths from you to the Target, and in the case of deep-water Charging, you should simply Charge _straight_ to your Target.
